BlueprintUI/Tests/AttributedTextTests.swift

```diff
@@ -0,0 +1,107 @@
+import XCTest
+@testable import BlueprintUI
+
+class AttributedTextTests: XCTestCase {
+
+    func testApplyingAttributes() {
+        let font = UIFont.systemFont(ofSize: 30)
+        let color = UIColor.green
+
+        var text = AttributedText(string: "Hello world")
+        text.font = font
+        text.color = color
+
+        text.attributedString.enumerateAttributes(
+            in: text.attributedString.entireRange,
+            options: []
+        ) { attributes, range, _ in
+            XCTAssertEqual(attributes[.font] as? UIFont, font)
+            XCTAssertEqual(attributes[.foregroundColor] as? UIColor, color)
+            XCTAssertEqual(range, text.attributedString.entireRange)
+        }
+    }
+
+    func testApplyingPartialAttributes() {
+        let font = UIFont.systemFont(ofSize: 30)
+        let green = UIColor.green
+        let blue = UIColor.blue
+
+        var text = AttributedText(string: "Hello world")
+
+        let hello = text.range(of: "Hello")!
+        let world = text.range(of: "world")!
+        let loWo = text.range(of: "lo wo")!
+        let rld = text.range(of: "rld")!
+
+        text[hello].font = font
+        text[world].color = green
+        text[loWo].color = blue
+
+        XCTAssertEqual(text[hello].font, font, "The font should be returned when it applies to the whole range")
```

Comment:
Can you also test that a contained range like `"ell"` and a containing range like `"hello wo"` are both nil?

Reply:
Done (although "ell" returns the font)